### PR TITLE
clang-tidy: use default member init

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/image.cpp [hunk 1]
```diff
@@ -116,17 +116,9 @@ std::string pathOfFileUrl(const std::string& url) {
 namespace Exiv2 {
 Image::Image(ImageType type, uint16_t supportedMetadata, BasicIo::UniquePtr io) :
     io_(std::move(io)),
-    pixelWidth_(0),
-    pixelHeight_(0),
+
     imageType_(type),
-    supportedMetadata_(supportedMetadata),
-#ifdef EXV_HAVE_XMP_TOOLKIT
```
Comment (neheb) [inline src/image.cpp:123]: Now I remember why I didn't do this before...

**Review by kevinbackhouse — DISMISSED** [review]

**Comment by piponazo** [comment]
PR looks good to me, although there is a clang-format issue.

**Comment by neheb** [comment]
I assume I'm using a newer version of clang-format that does not behave the same.

**Comment by neheb** [comment]
Yeah.

```
clang-format version 14.0.0 (Fedora 14.0.0-1.fc36)
```

results in

```
--- a/src/image_int.hpp
+++ b/src/image_int.hpp
@@ -65,7 +65,7 @@ struct binaryToStringHelper {
   explicit binaryToStringHelper(const Slice<T> myBuf) noexcept : buf_(myBuf) {
   }

-  friend std::ostream& operator<<<T>(std::ostream& stream, const binaryToStringHelper<T>& binToStr);
+  friend std::ostream& operator<< <T>(std::ostream& stream, const binaryToStringHelper<T>& binToStr);

   // the Slice is stored by value to avoid dangling references, in case we
   // invoke:
```

**Comment by neheb** [comment]
diff between 11 and 14 for google style.

```
colordiff 11 14
6,9c6,10
< AlignConsecutiveMacros: false
< AlignConsecutiveAssignments: false
< AlignConsecutiveBitFields: false
< AlignConsecutiveDeclarations: false
---
> AlignArrayOfStructures: None
> AlignConsecutiveMacros: None
> AlignConsecutiveAssignments: None
> AlignConsecutiveBitFields: None
> AlignConsecutiveDeclarations: None
14d14
< AllowAllConstructorInitializersOnNextLine: true
26a27,28
> AttributeMacros:
>   - __capability
48a51
> BreakBeforeConceptDeclarations: true
58a62
> QualifierAlignment: Leave
60d63
< ConstructorInitializerAllOnOneLineOrOnePerLine: true
66a70,71
> EmptyLineAfterAccessModifier: Never
> EmptyLineBeforeAccessModifier: LogicalBlock
67a73,76
> PackConstructorInitializers: NextLine
> BasedOnStyle:    ''
> ConstructorInitializerAllOnOneLineOrOnePerLine: false
> AllowAllConstructorInitializersOnNextLine: true
72a82,83
> IfMacros:
>   - KJ_IF_MAYBE
77a89
>     CaseSensitive:   false
80a93
>     CaseSensitive:   false
83a97
>     CaseSensitive:   false
86a101
>     CaseSensitive:   false
88a104
> IndentAccessModifiers: false
93a110
> IndentRequires:  false
99a117
> LambdaBodyIndentation: Signature
112a131
> PenaltyBreakOpenParenthesis: 0
116a136
> PenaltyIndentedWhitespace: 0
117a138
> PPIndentWidth:   -1
146c167
<     CanonicalDelimiter: ''
---
>     CanonicalDelimiter: pb
147a169
> ReferenceAlignment: Pointer
149c171,175
< SortIncludes:    true
---
> RemoveBracesLLVM: false
> SeparateDefinitionBlocks: Leave
> ShortNamespaceLines: 1
> SortIncludes:    CaseSensitive
> SortJavaStaticImport: Before
154a181
> SpaceBeforeCaseColon: false
158a186,194
> SpaceBeforeParensOptions:
>   AfterControlStatements: true
>   AfterForeachMacros: true
>   AfterFunctionDefinitionName: false
>   AfterFunctionDeclarationName: false
>   AfterIfMacros:   true
>   AfterOverloadedOperator: false
>   BeforeNonEmptyParentheses: false
> SpaceAroundPointerQualifiers: Default
163c199
< SpacesInAngles:  false
---
> SpacesInAngles:  Never
166a203,205
> SpacesInLineCommentPrefix:
>   Minimum:         1
>   Maximum:         -1
169a209
> BitFieldColonSpacing: Both
170a211,212
> StatementAttributeLikeMacros:
>   - Q_EMIT
180a223,224
>   - NS_SWIFT_NAME
>   - CF_SWIFT_NAME
```

**Review by kevinbackhouse — APPROVED** [review]

**Review by piponazo — APPROVED** [review]

**Pull request merged**